crates/superposition/src/webhooks/handlers.rs

```diff
@@ -0,0 +1,158 @@
+use actix_web::{
+    delete, get, post, put,
+    web::{self, Json, Query},
+    HttpResponse, Scope,
+};
+use chrono::Utc;
+use service_utils::service::types::{DbConnection, SchemaName};
+use superposition_types::database::models::others::WebhookVersion;
+use superposition_types::{
+    api::webhook::{CreateWebhookRequest, UpdateWebhookRequest, WebhookName},
+    custom_query::PaginationParams,
+    result as superposition, PaginatedResponse, User,
+};
+
+use super::helper::{fetch_webhook, validate_events};
+use diesel::{delete, ExpressionMethods, QueryDsl, RunQueryDsl};
+use superposition_types::database::models::others::Webhook;
+use superposition_types::database::schema::webhook::{self, dsl};
+pub fn endpoints() -> Scope {
+    Scope::new("")
+        .service(create)
+        .service(list_webhooks)
+        .service(get)
+        .service(update)
+        .service(delete_webhook)
+}
+
+#[post("")]
+async fn create(
+    request: Json<CreateWebhookRequest>,
+    db_conn: DbConnection,
+    schema_name: SchemaName,
+    user: User,
+) -> superposition::Result<Json<Webhook>> {
+    let DbConnection(mut conn) = db_conn;
+    let req = request.into_inner();
+
+    validate_events(&req.events, None, &schema_name, &mut conn)?;
+    let now = Utc::now();
+    let create_webhook = Webhook {
+        name: req.name,
+        description: req.description,
+        enabled: req.enabled,
+        url: req.url,
+        method: req.method,
+        version: req.version.unwrap_or(WebhookVersion::V1),
```

Comment:
if we are planning to keep this in the request, can we use `Default` trait here, because when someone would be configuring at a later point in time, the Default value we can just make sure is always the newer value

```suggestion
        version: req.version.unwrap_or_default(),
```